src/ui/view/dataverse/component/dynamicCheckbox/dynamicCheckbox.tsx

```diff
@@ -0,0 +1,67 @@
+import type { FC } from 'react'
+import { useCallback } from 'react'
+import classNames from 'classnames'
+import { Checkbox } from '@/ui/component/checkbox/checkbox'
+import { TextHighlighter } from '@/ui/component/textHighlighter/textHighlighter'
+import type { SelectionOption } from '@/ui/view/dataverse/component/filters/selectionFilter/selectionFilter'
+import './dynamicCheckbox.scss'
+
+type DynamicCheckboxProps = {
+  name: string
+  value: string
+  checked: boolean
+  highlightedTerm: string
+  disabled?: boolean
+  onCheckedChange: (checkboxOption: Pick<SelectionOption, 'selected' | 'value' | 'name'>) => void
```

Comment:
I think you can update with an `Omit` ;)

Reply:
The goal of using `Pick` is to explicitly show which props from the checkbox are necessary

Reply:
Works for me !